src/components/Garden/Preferences/EVMExecutor.js

```diff
@@ -0,0 +1,207 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import {
+  Box,
+  Button,
+  DropDown,
+  Field,
+  GU,
+  Info,
+  TextInput,
+} from '@1hive/1hive-ui'
+import { EVMcrispr } from '@1hive/evmcrispr'
+
+import MultiModal from '@components/MultiModal/MultiModal'
+import CreateDecisionScreens from '../ModalFlows/CreateDecisionScreens/CreateDecisionScreens'
+
+import { useGardens } from '@/providers/Gardens'
+import { useWallet } from '@/providers/Wallet'
+
+import { SHORTENED_APPS_NAMES } from '@utils/app-utils'
+// import { asciiToHex } from '@utils/web3-utils'
+
+import actions from '../../../actions/garden-action-types'
+import radspec from '../../../radspec'
+
+function EVMExecutor() {
+  const { account, ethers } = useWallet()
+  const { connectedGarden } = useGardens()
+  const [createDecisionModalVisible, setCreateDecisionModalVisible] = useState(
+    false
+  )
+  const [evmcrispr, setEvmcrispr] = useState(null)
+  const [installedApps, setInstalledApps] = useState([])
+  const [selectedApp, setSelectedApp] = useState(null)
+  const [selectedFunction, setSelectedFunction] = useState(null)
+  const [parameters, setParameters] = useState([])
+
+  useEffect(() => {
+    async function getEvmCrispr() {
+      if (!connectedGarden) {
+        return
+      }
+      const crispr = await EVMcrispr.create(
+        connectedGarden.address,
+        ethers.getSigner()
+      )
+
+      setEvmcrispr(crispr)
+    }
+    getEvmCrispr()
+  }, [connectedGarden, ethers])
+
+  useEffect(() => {
+    if (!evmcrispr) {
+      return
+    }
+    const apps = evmcrispr.apps()
```

Comment:
Since this is not something returned async, couldn´t we just use a `useMemo` here? Like we do for the functions for example.